conans/test/unittests/client/build/cpp_std_flags_test.py

```diff
@@ -1,195 +1,261 @@
-        self.assertEqual(cppstd_flag("clang", "8", "17"), '-std=c++17')
-        self.assertEqual(cppstd_flag("clang", "8", "20"), '-std=c++2a')
+        with self.assertRaises(ConanException):
+            make_cppstd_default("gcc", "4.0")
+        self.assertEqual(make_cppstd_default("gcc", "5"), "gnu98")
+        self.assertEqual(make_cppstd_default("gcc", "6"), "gnu14")
+        self.assertEqual(make_cppstd_default("gcc", "6.1"), "gnu14")
+        self.assertEqual(make_cppstd_default("gcc", "7.3"), "gnu14")
+        self.assertEqual(make_cppstd_default("gcc", "8.1"), "gnu14")
+
+    def test_clang_make_cppstd_flags(self):
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "2.0", "98")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "2.0", "gnu98")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "2.0", "11")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "2.0", "14")
+
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "2.1", "98")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "2.1", "gnu98")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "2.1", "11")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "2.1", "14")
+
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "3.0", "11")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "3.0", "14")
+
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "3.1", "11")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "3.1", "14")
+
+        self.assertEqual(make_cppstd_flag("clang", "3.4", "11"), '-std=c++11')
+        self.assertEqual(make_cppstd_flag("clang", "3.4", "14"), '-std=c++1y')
+        self.assertEqual(make_cppstd_flag("clang", "3.4", "17"), None)
+
+        self.assertEqual(make_cppstd_flag("clang", "3.5", "11"), '-std=c++11')
+        self.assertEqual(make_cppstd_flag("clang", "3.5", "14"), '-std=c++14')
+        self.assertEqual(make_cppstd_flag("clang", "3.5", "17"), '-std=c++1z')
+
+        self.assertEqual(make_cppstd_flag("clang", "5.0", "11"), '-std=c++11')
+        self.assertEqual(make_cppstd_flag("clang", "5.0", "14"), '-std=c++14')
+        self.assertEqual(make_cppstd_flag("clang", "5.0", "gnu14"), '-std=gnu++14')
+        self.assertEqual(make_cppstd_flag("clang", "5.0", "17"), '-std=c++17')
+
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "5.1", "11")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "5.1", "14")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("clang", "5.1", "17")
```

Comment:
Failing because of the helper function, not because of the underlying `cppstd_flag` being tested here

Reply:
removed